bridge-sdk/src/nep141/mod.rs

```diff
@@ -29,50 +29,62 @@ abigen!(
+    eth_chain_id: Option<u64>,
     eth_private_key: Option<String>,
-    bridge_token_factory_address: String,
+    bridge_token_factory_address: Option<String>,
     near_endpoint: Option<String>,
     near_private_key: Option<String>,
     near_signer: Option<String>,
-    token_locker_id: String,
-    environment: Env,
+    token_locker_id: Option<String>,
+    near_light_client_address: Option<String>,
 }
 
 impl Nep141Bridging {
-    /// Creates a new instance of the bridging client. Constant values are set here, while the others can be provided separately depending on the use case.
-    pub fn new(env: Env) -> Self {
-        match env {
-            Env::Testnet => Self {
-                eth_chain_id: 11155111,
-                bridge_token_factory_address: "0xa9108f7F83Fb661e611991116D526fCa1a9585ab".to_string(),
-                eth_endpoint: None,
-                eth_private_key: None,
-                near_endpoint: None,
-                near_private_key: None,
-                near_signer: None,
-                token_locker_id: "ft-locker.sepolia.testnet".to_string(),
-                environment: env,
-            },
-            Env::Mainnet => Self {
-                eth_chain_id: 1,
-                bridge_token_factory_address: "0x252e87862A3A720287E7fd527cE6e8d0738427A2".to_string(),
-                eth_endpoint: None,
-                eth_private_key: None,
-                near_endpoint: None,
-                near_private_key: None,
-                near_signer: None,
-                token_locker_id: "ft-locker.bridge.near".to_string(),
-                environment: env,
-            }
+    /// Creates an empty instance of the bridging client. Property values can be set separately depending on the required use case.
+    pub fn new() -> Self {
+        Self {
+            eth_chain_id: None,
+            bridge_token_factory_address: None,
+            eth_endpoint: None,
+            eth_private_key: None,
+            near_endpoint: None,
+            near_private_key: None,
+            near_signer: None,
+            token_locker_id: None,
+            near_light_client_address: None,
         }
     }
 
+    /// Set bridge token factory address on Ethereum
+    /// 
+    /// Required for `deploy_token`, `mint`, `burn`
+    pub fn with_bridge_token_factory_address(mut self, address: String) -> Self {
+        self.bridge_token_factory_address = Some(address);
+        self
+    }
+
+    /// Set token locked account id on Near
+    /// 
+    /// Required for `log_token_metadata`, `storage_deposit_for_token`, `deploy_token`, `deposit`, `mint`, `withdraw`
+    pub fn with_token_locker_id(mut self, token_locker_id: String) -> Self {
+        self.token_locker_id = Some(token_locker_id);
+        self
+    }
+
+    /// Set Near light client address on Ethereum
+    /// 
+    /// Required for `deploy_token`, `mint`
+    pub fn with_near_light_client_address(mut self, light_client_address: String) -> Self {
```

Comment:
Please replace this setters with [derive_builder](https://docs.rs/derive_builder/latest/derive_builder/).